client/src/components/R&R/NewReview.jsx

```diff
@@ -4,72 +4,279 @@ import $ from 'jquery';
 import axios from 'axios';
 import styled from 'styled-components';
 import handler from '../Shared/reviewhandler.js';
-
-
+import StarRating from './InteractiveStars.jsx';
+import NewReviewChars from './NewReviewChars.jsx';
+import Errors from './Errors.jsx';
 
 class NewReview extends React.Component {
   constructor(props) {
     super(props);
     this.state = {
       show: false,
       stars: null,
-      recommend: null,
-      characteristics: null,
+      recommend: true,
+      characteristics: {},
       summary: '',
       body: '',
       images: [],
       nickname: '',
       email: '',
-      count: 60,
+      count: `Minimum required characters left: 50`,
+      imgWindow: false,
+      submitted: false,
+      errors: {},
+      showError: false
     };
+
+    this.handleInputChange = this.handleInputChange.bind(this);
     this.toggleModal = this.toggleModal.bind(this);
+    this.toggleImgWindow = this.toggleImgWindow.bind(this);
     this.submitReview = this.submitReview.bind(this);
+    this.characterCounter = this.characterCounter.bind(this);
+    this.processCharSelection = this.processCharSelection.bind(this);
+  }
+
+  characterCounter(e) {
+    let charsLeft = 50 - e.target.value.length;
```

Comment:
oof, what if `e.target.value` is undefined? you may want to handle a default scenario

`(e?.target?.value || "").length`